### PR TITLE
Improve event typing and update `web3` for the `web3-1.0.0` target

### PR DESCRIPTION
**Pull request by JamesLefrere**
Hey @krzkaczor , this is a great project! I had a go at improving the types for events, so that event listeners/event callbacks have properly typed params:

Screenshot 2019-05-03 at 20 54 52

While I was at it, I updated the `web3` version; the biggest change here is probably the BigNumber change in web3: https://github.com/ethereum/web3.js/issues/2570 

**Changes**

* Upgrade the `web3-1.0.0` target's `web3` version to the latest beta (54)
* Adjust test setup and assertions to reflect `web3` changes
* Parse event inputs in order to improve the typing of contract events for the `web3-1.0.0` target
* Update existing types to reflect updated `web3` (e.g. `EstimateGasOptions`)

Let me know what you think. Cheers!

### CODE REVIEW

File: test/integration/targets/web3-1.0.0/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   },
   "scripts": {
     "tsc": "tsc --noEmit",
-    "mocha": "NODE_ENV=test mocha --require ts-node/register.js './**/*.spec.web3.ts'",
+    "mocha": "NODE_ENV=test mocha --require ts-node/register.js './**/*.spec.web3.ts' --timeout 5000",
```
Comment (JamesLefrere) [inline test/integration/targets/web3-1.0.0/package.json]: Some of the tests took over 2s (Mocha's default) on my computer.

File: test/integration/targets/web3-1.0.0/web3.ts [hunk 2]
```diff
@@ -10,7 +10,9 @@ export let web3: Web3;
 export let accounts: string[];
 
 export async function createNewBlockchain() {
-  const web3 = new Web3(ganache.provider());
+  const web3 = new Web3(ganache.provider(), undefined, {
+    transactionConfirmationBlocks: 1,
```
Comment (JamesLefrere) [inline test/integration/targets/web3-1.0.0/web3.ts]: I didn't look too deeply into this, but without this option, Web3 seemed to hang around for more confirmations.

File: test/integration/targets/web3-1.0.0/ContractWithOverloads.spec.web3.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-import { DumbContract } from "./types/web3-contracts/DumbContract";
+import { ContractWithOverloads } from "./types/web3-contracts/ContractWithOverloads";
 
 import { expect } from "chai";
-import { ContractWithOverloads } from "./types/web3-contracts/ContractWithOverloads";
 
 describe("ContractWithOverloads", () => {
   it("should work", async () => {
-    const contract = (await deployContract("ContractWithOverloads")) as ContractWithOverloads;
+    const contract = await deployContract<ContractWithOverloads>("ContractWithOverloads");
```
Comment (JamesLefrere) [inline test/integration/targets/web3-1.0.0/ContractWithOverloads.spec.web3.ts]: I don't know which style is preferred; this is just like the other test now.

File: test/integration/targets/web3-1.0.0/web3.ts (hunk 2 above)
Comment (krzkaczor) [reply test/integration/targets/web3-1.0.0/web3.ts]: Hmm i saw similar behaviour with *SOME* web3 versions in the past.

File: test/integration/targets/web3-1.0.0/DumbContract.spec.web3.ts [hunk 4]
```diff
@@ -23,16 +22,16 @@ describe("DumbContract", () => {
...
-    expect(await contract.methods.counter().call()).to.be.eq("4");
+    expect((await contract.methods.counter().call()).toString()).to.be.eq("4");
   });
 
   it("should allow to pass signed values in multiple ways", async () => {
     const contract = await deployContract<DumbContract>("DumbContract");
 
-    expect(await contract.methods.returnSigned(2).call()).to.be.eq("2");
-    expect(await contract.methods.returnSigned("2").call()).to.be.eq("2");
+    expect((await contract.methods.returnSigned(2).call()).toString()).to.be.eq("2");
```
Comment (krzkaczor) [inline test/integration/targets/web3-1.0.0/DumbContract.spec.web3.ts]: according to new typings this should be BN, right? Could you add some kind of type assertion as well?

Comment (JamesLefrere) [reply test/integration/targets/web3-1.0.0/DumbContract.spec.web3.ts]: Good idea, I've now done this in [44e6062](https://github.com/ethereum-ts/TypeChain/commit/44e6062a500abcc46af0562736f42367bf243f53) .

It looks like there's still some BigNumber/BN confusion to sort out in web3: https://github.com/ethereum/web3.js/issues/2468

**Comment by krzkaczor** [comment]
Merged! sorry that it took so long!

**Pull request merged**